### DISCUSSION TITLE
Licensing question for extending Prism functionality

### DISCUSSION THREAD

**Discussion by mr5z** [PrismLibrary/Prism#3037] [.NET MAUI] [answered]
### Background
I have created a library which extends some `INavigationService` functionality and [Mopups](https://github.com/LuckyDucko/Mopups) integration (closely similar to [Prism.Plugin.Popups](https://github.com/dansiegel/Prism.Plugin.Popups)).
### Question
Which license is compatible with Prism if I want to publish some public repo & NuGet package to extend Prism capabilities?

**Comment by dansiegel** [comment, marked as answer]
Essentially what you are asking would violate section 4.2(e) of the Prism License. We already ship a version of Prism.Plugin.Popups for Mopups which is available to our Commercial Plus License holders on the private Prism NuGet feed.

While you are welcome to write libraries that extend Prism's functionality for your own internal use, publication of those libraries would constitute a breach of the license.

**Reply by mr5z** [reply]
Got it. Thanks for quick response.
